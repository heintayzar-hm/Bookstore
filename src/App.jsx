import React from 'react';
import { Route, Routes } from 'react-router-dom';
import Layout from './component/layout/Layout';
import BooksPage from './pages/BookPage';
import CategoriesPage from './pages/CategoriesPage';

// eslint-disable-next-line react/prefer-stateless-function
class App extends React.Component {
  constructor(props) {
    super();
    this.props = props;
  }

  render() {
    return (
      <>
        <Layout>
          <Routes>
            <Route path="/books" element={<BooksPage />} />
            <Route path="/categories" element={<CategoriesPage />} />
            <Route path="/*" element={<BooksPage />} />
          </Routes>
        </Layout>
      </>
    );
  }
}
export default App;
