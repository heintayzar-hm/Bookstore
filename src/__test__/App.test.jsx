import { describe, test } from 'vitest';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { HashRouter } from 'react-router-dom';
import App from '../App';
import store from '../state/store';

describe('App', () => {
  test('renders the initial state correctly', () => {
    // Arrange

    // Act
    const { container } = render(
      <HashRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </HashRouter>,
    );
    // Assert
    expect(container).toMatchSnapshot();
  });
});
