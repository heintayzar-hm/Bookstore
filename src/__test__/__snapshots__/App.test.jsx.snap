// Vitest Snapshot v1

exports[`App > renders the initial state correctly 1`] = `
<div>
  <nav>
    <h1>
      BookStore
    </h1>
    <div>
      <ul>
        <li>
           
          <a
            class=""
            data-testid="link-item"
            href="#/books"
          >
            Books
          </a>
        </li>
        <li>
           
          <a
            class=""
            data-testid="link-item"
            href="#/categories"
          >
            Categories
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <ul />
  <article
    data-testid="form-container"
  >
    <h2>
      Add New Book
    </h2>
    <form>
      <input
        data-testid="book-title-input"
        name="title"
        placeholder="Book Title"
        required=""
        type="text"
        value=""
      />
      <input
        data-testid="author-title-input"
        name="author"
        placeholder="Author"
        required=""
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add Book
      </button>
    </form>
  </article>
</div>
`;
